### ISSUE TITLE
Update branches immediately to match target

### ISSUE THREAD

**Issue by lucleray** [chdsbd/kodiak#120]
Hi 👋

Currently, PR branches are updated once all the other checks pass (approvals, label, ...).

I'd like to suggest updating the branch even when all those checks don't pass. The reason behind that is that we want to know that our branch is breaking as soon as possible. If we wait for all the other checks to pass, we might be losing time reviewing a branch that is broken once master in merged into it (and therefore will have to review it twice).

Do you think it is possible to implement this into Kodiak?

Thank you for your help 🙏

**Comment by chdsbd** [comment]
I've talked about this a little bit before in https://github.com/chdsbd/kodiak/issues/75, but let me expand on that.

Kodiak tries to be efficient when updating branches, so it will only update a branch when it's about to be merged into the target branch. This is by design to reduce useless updates to other branches.

We could definitely add this feature, the challenge I see is that whenever the target branch (usually master) is updated, all the PRs against it will need updates, which may not be desired.

Let me know if this is still a feature you'd like to see in Kodiak. It shouldn't be too hard to add.

**Comment by watchinharrison** [comment]
I can understand the inefficiency in this but it would be nice to opt into this functionality, if at all possible

**Comment by chdsbd** [comment]
@watchinharrison I agree. This is a planned feature and I should have time to work on it after #96 is fixed.

**Comment by lucleray** [comment]
@chdsbd I see this section in the README: https://github.com/chdsbd/kodiak#auto-merging

Is it implemented? How do I activate it? 😃

**Comment by chdsbd** [comment]
@lucleray the feature described in this PR hasn’t been implemented yet. To enable kodiak right now please follow the setup steps here: https://github.com/chdsbd/kodiak#setup

By default, once you’ve installed the GitHub app you need to add a kodiak configuration file and label your PRs with a label (by default “automerge”) to enable kodiak to automatically merge the PR.

**Comment by lucleray** [comment]
Sorry, wrong link! I meant to link this section: "Auto-Update branches"

(https://github.com/chdsbd/kodiak#auto-update-branches)

**Comment by chdsbd** [comment]
@lucleray Thanks. Kodiak only updates a PR when it's up for merging at the moment, so this issue still needs to be addressed to enable updating a PR unconditionally.

**Comment by lucleray** [comment]
@chdsbd Ok 🙂Thanks for the answer!

**Comment by chdsbd** [comment]
This feature has been added and deployed to production.

**Comment by lucleray** [comment]
Thank you @chdsbd!
